receiver/pluginreceiver/receiver.go

```diff
@@ -83,12 +105,9 @@ func (r *Receiver) Shutdown(ctx context.Context) error {
 
 // startService starts the provided service
 func (r *Receiver) startService(ctx context.Context, svc Service) error {
-	errChan := make(chan error)
 	go func() {
-		if err := svc.Run(ctx); err != nil {
-			errChan <- err
-		}
-		close(r.doneChan)
+		r.serviceErrChan <- svc.Run(ctx)
+		close(r.serviceErrChan)
 	}()
```

Comment:
this is where the problem was; previously, errChan was scoped to this method, and once this method exited nothing would be able to listen to that channel, which could cause this goroutine to block indefinitely.